resources/assets/components/Signup/index.js

```diff
@@ -304,6 +311,14 @@ class Signup extends React.Component {
   // @TODO: make this work for any type of post?
   // Submit a new photo post on behalf of the user.
   submitPost(post) {
+    // To submit a post with a file, we need to change the Content-Type to muultipart/form-data.
+    this.api = new RestApiClient(window.location.origin, {
```

Comment:
Rather than swapping out the shared client, let's store this in a local variable `const api = ...` here (otherwise this would be end up getting used by subsequent calls elsewhere).

Reply:
oh smart!!